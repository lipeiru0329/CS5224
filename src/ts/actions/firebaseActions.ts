import * as CST from 'ts/common/constants';
import { VoidThunkAction } from 'ts/common/types';

export function authUpdate(signedIn: boolean) {
	return {
		type: CST.AC_AUTH,
		[CST.AC_AUTH]: signedIn
	};
}

export function updateResult(result: any[]): VoidThunkAction {
	console.log(result);
	console.log(';;;;;;;;;;;;;;;');
	return async dispatch => {
		dispatch(resultUpdating(result));
	};
}

export function updateDetail(data: any) {
	return {
		type: CST.AC_DETAIL,
		[CST.AC_DETAIL]: data
	};
}

export function resultUpdating(result: any[]) {
	return {
		type: CST.AC_RESULT,
		[CST.AC_RESULT]: result
	};
}

export function foodUpdating(result: any) {
	return {
		type: CST.AC_FOOD,
		[CST.AC_FOOD]: result
	};
}
