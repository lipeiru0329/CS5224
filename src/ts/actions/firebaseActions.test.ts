import * as firebaseActions from './firebaseActions';

describe('actions', () => {
	test('authUpdate', () => {
		expect(firebaseActions.authUpdate(true)).toMatchSnapshot();
	});
});
