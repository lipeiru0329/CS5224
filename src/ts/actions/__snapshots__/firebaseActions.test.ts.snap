// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`actions authUpdate 1`] = `
Object {
  "auth": true,
  "type": "auth",
}
`;
